nifi-stateless/nifi-stateless-api/src/main/java/org/apache/nifi/stateless/flow/TriggerResult.java

```diff
@@ -57,11 +58,18 @@
      * Provides the contents of a FlowFile that was obtained by calling {@link #getOutputFlowFiles()}.
      * @param flowFile the FlowFile whose contents are to be read
      * @return the contents of the FlowFile
+     * @throws IOException if unable to read the contents of the FlowFile
      */
-    byte[] readContent(FlowFile flowFile);
+    byte[] readContent(FlowFile flowFile) throws IOException;
```

Comment:
If throwing an `IOException` can only occur in the file system implementation, would it make sense to keep the existing method signature and change the implementing classes to wrap `IOException` in `UncheckedIOException`? It depends on how likely it is for an IOException to occur, and whether the caller should be expected to do anything useful with the IOException. Just something to consider, this may be best as implemented.

Reply:
Thanks for the explanation, having the declared IOException as part of the interface makes sense going forward.